packages/nextjs/src/app-router/server/auth.ts

```diff
@@ -1,12 +1,78 @@
+import type { SignedInAuthObject, SignedOutAuthObject } from '@clerk/backend';
+import type { CheckAuthorizationWithCustomPermissions } from '@clerk/types';
+import { notFound } from 'next/navigation';
+
 import { authAuthHeaderMissing } from '../../server/errors';
 import { buildClerkProps, createGetAuth } from '../../server/getAuth';
+import type { AuthObjectWithoutResources } from '../../server/types';
 import { buildRequestLike } from './utils';
 
+type AuthSignedIn = AuthObjectWithoutResources<
+  SignedInAuthObject & {
+    /**
+     * @experimental
+     * This function is experimental as it throws a Nextjs notFound error if user is not authenticated or authorized
+     */
+    protect: (
+      params?:
+        | Parameters<CheckAuthorizationWithCustomPermissions>[0]
+        | ((has: CheckAuthorizationWithCustomPermissions) => boolean),
+    ) => AuthObjectWithoutResources<SignedInAuthObject>;
+  }
+>;
+
+type AuthSignedOut = AuthObjectWithoutResources<
+  SignedOutAuthObject & {
+    /**
+     * @experimental
+     * This function is experimental as it throws a Nextjs notFound error if user is not authenticated or authorized
+     */
+    protect: never;
+  }
+>;
+
 export const auth = () => {
-  return createGetAuth({
+  const authObject = createGetAuth({
     debugLoggerName: 'auth()',
     noAuthStatusMessage: authAuthHeaderMissing(),
   })(buildRequestLike());
+
+  (authObject as AuthSignedIn).protect = params => {
```

Comment:
We talked about supporting `redirectUrl` as an optional parameter when doing authz checks, but I can't remember if we decided to omit that for the initial implementation 🤔

```ts
auth().protect({ role: 'admin' }, { redirectUrl: '/home' })
```

Reply:
I think we are skipping this. Especially with something like `notAuthorized` we wouldn't need to support `redirectUrl`